sw/device/lib/dif/dif_rv_timer.c

```diff
@@ -158,6 +159,36 @@ dif_rv_timer_result_t dif_rv_timer_counter_read(const dif_rv_timer_t *timer,
   }
 }
 
+dif_rv_timer_result_t dif_rv_timer_counter_write(const dif_rv_timer_t *timer,
+                                                 uint32_t hart_id,
+                                                 uint64_t count) {
+  if (timer == NULL || hart_id >= timer->config.hart_count) {
+    return kDifRvTimerBadArg;
+  }
+
+  // Disable the counter.
+  uint32_t ctrl_reg =
+      mmio_region_read32(timer->base_addr, RV_TIMER_CTRL_REG_OFFSET);
+  uint32_t ctrl_reg_cleared = bitfield_bit32_write(ctrl_reg, hart_id, false);
+  mmio_region_write32(timer->base_addr, RV_TIMER_CTRL_REG_OFFSET,
+                      ctrl_reg_cleared);
+
+  // Write the new count.
+  uint32_t lower_count = (uint32_t)count;
```

Comment:
You shouldn't need these casts thanks to ~implicit conversions~, but you can leave them if you really want.

Reply:
fixed